### DISCUSSION TITLE
Stagnant PRs

### DISCUSSION THREAD

**Discussion by QuintillusCFC** [C7-Game/Prototype#317] [General]
I think we should probably adopt a policy about stagnant PRs.  Meaning, PRs that were opened (potentially as a draft), perhaps received comments, but were not resolved/approved and integrated for an extended period of time.

We have two currently, #235 and #204 , that haven't had any activity since May, and cover three issues-in-progress.  On number 235, a significant number of conflicts have emerged in the interim.  The features they cover - save/load and regression tests - are things we want to have in the game, sooner rather than later.

IIRC, these are idle because @pcen got busy with real life, which is to be expected.  I'm sure it won't be the last time, and IIRC we had the same thing happen to a couple of PRs around the end of December/start of January, which were eventually merged in March.

My proposal is that if someone hasn't responded to their PR in two weeks, and hasn't given any indication that they plan to come back to it within a given period of time, then anyone can pick it up and move it along so it can be merged.  In many cases these PRs just need minor changes, and it will be less work for someone to do the last 15% of them than for them to accumulate a greater many merge conflicts over a period of months.  And we don't want priority items to get stuck in limbo indefinitely, either.

Any input from Paul would be great (hope you're doing okay!), also @WildWeazel has the Component Event framework that's at 13 days since the last update.  Maybe 2 weeks is too soon, would 4 be better?  I don't want to have us stealing each others' PRs, but I don't want them sitting for 6 months, either.

**Comment by pcen** [comment]
Unfortunately I haven’t been able to contribute in the past few months, and likely won’t be able to for another few weeks. Having said that, I’d encourage anyone at anytime to push to any branch I have / contribute to any PR I’ve opened, so I’m in support of a 2 week policy!

**Comment by WildWeazel** [comment]
That sounds like a reasonable rule of thumb. Of course it's worth pinging them before picking it up in case they have local WIP or are still thinking through a problem.
